content/guides/component-testing/events-angular.md

```diff
@@ -0,0 +1,414 @@
+      change: {
+        emit: cy.spy().as('changeSpy'),
+      } as any,
+    },
+  })
+  // Act
+  cy.get('incrementSelector').click()
+})
+```
+
+</code-block>
+<code-block label="stepper.component.cy.ts (with Template)">
+
+```ts
+it('clicking + fires a change event with the incremented value', () => {
+  // Arrange
+  cy.mount('<app-stepper (change)="change.emit($event)"></app-stepper>', {
+    declarations: [StepperComponent],
+    componentProperties: {
+      change: {
+        emit: cy.spy().as('changeSpy'),
+      } as any,
+    },
+  })
+  // Act
+  cy.get('incrementSelector').click()
+})
+```
+
+</code-block>
+</code-group>
+
+### Assert
+
+Finally, we **Assert** that the `change` event was emitted with the correct
+value.
+
+<code-group>
+<code-block label="stepper.component.cy.ts" active>
+
+```ts
+it('clicking + fires a change event with the incremented value', () => {
+  // Arrange
+  cy.mount(StepperComponent, {
+    componentProperties: {
+      change: {
+        emit: cy.spy().as('changeSpy'),
+      } as any,
+    },
+  })
+  // Act
+  cy.get('incrementSelector').click()
+  // Assert
+  cy.get('@onChangeSpy').should('have.been.calledWith', 1)
+})
+```
+
+</code-block>
+<code-block label="stepper.component.cy.ts (with Template)">
+
+```ts
+it('clicking + fires a change event with the incremented value', () => {
+  // Arrange
+  cy.mount('<app-stepper (change)="change.emit($event)"></app-stepper', {
+    componentProperties: {
+      change: {
+        emit: cy.spy().as('changeSpy'),
+      } as any,
+    },
+  })
+  // Act
+  cy.get('incrementSelector').click()
+  // Assert
+  cy.get('@onChangeSpy').should('have.been.calledWith', 1)
+})
+```
+
+</code-block>
+</code-group>
+
+We may decide to combine this test with the previous tests we've written that
+test multiple things at once in a given scenario.
+
+Doing so is up to the discretion of the developer. Combining tests will result
+in a faster overall test run. However, it may be more challenging to isolate why
+a test failed in the first place. We recommend having longer tests for
+end-to-end tests because setup and visiting pages are expensive. Longer tests
+are not necessarily a problem for component tests because they are comparatively
+quick.
+
+## Using Cypress Promise API
+
+You can also use `.then()` which enables us to work with the subject that was
+yielded from the `cy.mount()` command. In this case, mount yields an object of
+`fixture`, and the `componentInstance` We can then use the `componentInstance`
+to create our `cy.spy()`
+
+This means that you are able to get to the resulting `response` returned from
+the `mount` command and use `response.component` in order to gain access to the
+properties of our component under test.
+
+Notice that we're using the `'should'` function signature in order to take
+advantage of Cypress's [retryability](/guides/guides/test-retries). If we
+chained using `should` instead of `cy.should`, we may run into the kinds of
```

Comment:
this sentence is a bit confusing because `cy.should()` is not a thing. Maybe you can rewrite this to be a bit more clear.